### ISSUE TITLE
Tedious initialisation using plugins

### ISSUE THREAD

**Issue by cprates** [micro/go-micro#1106]
Hi, I'm wondering if there is a simpler way to initialise a service while using plugins:

```
package main

import (
    "context"
    "log"

    "github.com/micro/go-micro"
    "github.com/micro/go-micro/server"
    _ "github.com/micro/go-plugins/server/grpc"
    _ "github.com/micro/go-plugins/transport/grpc"
)

func HandlerWrapper(handlerFunction server.HandlerFunc) server.HandlerFunc {
    return func(ctx context.Context, req server.Request, rsp interface{}) (err error) { 
        log.Println("Handling requet...")
        return
    }
}

func main() {
    service := micro.NewService(
        micro.Name("test_service"),
        micro.WrapHandler(HandlerWrapper),
    )

    log.Println("service name before parsing env vars:", service.Options().Server.Options().Name)

    service.Init(
        micro.Name("test_service"),
    )

    log.Println("service name after parsing env vars:", service.Options().Server.Options().Name)

    service.Server().Init(
        server.Name("test_service"),
        server.WrapHandler(HandlerWrapper),
    )

    log.Println("after re-setting service name:", service.Options().Server.Options().Name)

    // Run the server
    if err := service.Run(); err != nil {
        log.Fatalln(err)
    }
}
```

The code above is an example of how I'm initialising my service. Here I'm only passing args to micro.NewService and service.Init() just to make my point clearer.

I'm running this code with
`MICRO_SERVER=grpc go run main.go`

And the output is
```
2020/01/11 12:05:39 service name before parsing env vars: test_service
2020/01/11 12:05:39 service name after parsing env vars: go.micro.server
2020/01/11 12:05:39 after re-setting service name: test_service
2020/01/11 12:05:39 Server [grpc] Listening on [::]:60423
2020/01/11 12:05:39 Broker [http] Connected to [::]:60424
...
```

As expected, service.Init() replaces the server, forcing me to call service.Server().Init() after it.
So basically I have to call micro.NewService, service.Init and service.Server().Init, plus service.Client().Init because in the actual service I'm also using some wrappers for the client, and I'm also loading the client plugin.

I'm wondering is there is a way to initialise the service in fewer steps?
Also, looking to the code it doesn't seem necessary to import ` _ "github.com/micro/go-plugins/server/grpc"` and  `_ "github.com/micro/go-plugins/client/grpc"`. Are they really necessary when using these plugins?

I'm currently using go-micro v1.9.1

Thanks in advance.

**Comment by vtolstov** [comment]
why you need so complicated workflow? why not simple call NewService with all needed params?
P.S. Please also try on latest released version of go-micro

**Comment by asim** [comment]
We now use grpc by default so this should be a non issue. Please upgrade to v2.
